src/message_filters/__init__.py

```diff
@@ -113,11 +119,16 @@ def connectInput(self, f):
     def add(self, msg):
         if not hasattr(msg, 'header') or not hasattr(msg.header, 'stamp'):
             if not self.allow_headerless:
-                rospy.logwarn("Cannot use message filters with non-stamped messages. "
-                              "Use the 'allow_headerless' constructor option to "
-                              "auto-assign ROS time to headerless messages.")
+                rclpy.logging._root_logger.log("can not use message filters"
```

Comment:
Please use the node's public API to get the logger not the modules implementation. 

Here's an example: https://github.com/ros2/demos/blob/e5ea2e7fec944d08d04816ef0b8e2f89e5171880/demo_nodes_py/demo_nodes_py/topics/talker.py#L34

Also this has a lot of reflow that would be good to avoid.

Reply:
@tfoote Actually, I've ever considered to use `get_logger()` of node, but there is nowhere to involve ros2 node or no need to extend class `Cache` from ros2 node of which allows to use `get_logger()`, that's why I use the root logger instead, thanks !

Reply:
I'm sorry I don't understand your statement that there's nowhere to involve a ros2 node, you have node `self.node` used on line 78 above. 

And as the `_root_logger` is prefixed by an underscore it's considered a private implementation detail you should not be directly calling it.

Reply:
@tfoote the self.node used on line 78 above is in another `class Subscriber` but not for `class Cache`, so in order to use get_logger from a node within `class Cache`, it's required to pass a parameter of "ros2 node" into `class Cache` or have "class Cache extend from ros2 node class", but if the involving of ros2 node here is merely for logging, do you think it's really necessary ? thanks.